tests/ixia/test_pfc_pause_lossless.py

```diff
@@ -0,0 +1,289 @@
+    7. From Rx port send pause frames on priorities either m or n. Such that
+       priority of 'Test Data Traffic' at Tx end == Pause Priority at Rx end.
+    8. You may repeat the steps 6 and 7 several times.
+    9. Expected result -
+       a. No 'Test Data Traffic' will not flow. Since priority of
+          that is always equal to the priority pause frame priority.
+       b. 'Background Data Traffic' will always flow.
+    10. Configure 'Test Data Traffic' such that it contains traffic items
+        of both the priorities m and n.
+    11. Configure 'Background Data Traffic' it contains all other traffic
+        except priorities m and n.	
+    12. From Rx port send pause frames on priorities both m and n.
+    13. When pause frames are started 'Test Data Traffic' will stop; 
+        and when pause frames are stopped 'Test Data Traffic' will start.
+    14. 'Background Data Traffic' will always flow.
+    15. Repeat the steps 10, to 14 several times.
+
+    Note: We are using default PFC enabled class of SONiC DUT as m and n. 
+          So, m = 3, n = 4.
+    """ 
+    port_list = list()
+    fanout_devices = IxiaFanoutManager(fanout_graph_facts)
+    fanout_devices.get_fanout_device_details(device_number=0)
+    device_conn = conn_graph_facts['device_conn']
+
+    for intf in fanout_devices.get_ports():
+        peer_port = intf['peer_port']
+        intf['speed'] = int(device_conn[peer_port]['speed'])
+        port_list.append(intf)
+
+    # The topology should have at least two interfaces.
+    pytest_assert(len(device_conn) >= 2,
+        "The topology should have at least two interfaces")
+
+    # Test pausing each lossless priority individually.
+    # Step - 6, 7, 8, and 9.
+    session = ixia_api_server_session
+    for prio in lossless_prio_dscp_map:
+        for i in range(len(port_list)):
+            vports = configure_ports(session, port_list)
+
+            rx_id = i
+            tx_id = (i + 1) % len(port_list)
+
+            rx_port = vports[rx_id]
+            tx_port = vports[tx_id]
+            rx_port_bw = port_list[rx_id]['speed']
+            tx_port_bw = port_list[tx_id]['speed']
+
+            pytest_assert(rx_port_bw == tx_port_bw)
+
+            # All the DSCP values mapped to this priority.
+            test_dscp_list = lossless_prio_dscp_map[prio]
+            # The other DSCP values.
+            bg_dscp_list = [x for x in range(64) if x not in test_dscp_list]
+
+            exp_dur = 2
+
+            run_pfc_exp(session=session,
+                        dut=duthost,
+                        tx_port=tx_port,
+                        rx_port=rx_port,
+                        port_bw=tx_port_bw,
+                        test_prio_list=[prio],
+                        test_dscp_list=test_dscp_list,
+                        bg_dscp_list=bg_dscp_list,
+                        exp_dur=exp_dur,
+                        test_traffic_pause_expected=True)
+
+            clean_configuration(session=session)
+
```

Comment:
Can we split this into two functions?

Reply:
This comment is implemented. I have splatted  the main function into two function as below:
test_pfc_pause_single_lossless_priority
test_pfc_pause_multi_lossless_priorities